v11 fixes

josueBulle opened the pull request:
- Fix logging feature port to v11 in cms_form
- file upload in cms_form
- super() in python 3
- Fix warning: use hasclass() instead of @class in xml xpath
- Fix advanced_translation migration:
  - Refactor buggy and complex get_date and added some tests
  - Fixed translations in child_switzerland, babel does not use the same patterns as strftime

josueBulle commented:
finally, I'm not sure it is a good idea to move the translations lines in csv (but they still need the fixes)

ecino reviewed (changes requested):
The tests are broken.

josueBulle commented:
I think the tests fail on the branch v11 too, it might not be related to this pr

josueBulle committed:
Fixes

Fix warning: use hasclass() instead of @class in xml
Fix translatable_model method's get_date
Fix advanced_translation
lints
Fixes
